docs/modules/angular/pages/ngrx/ngrx.adoc

```diff
@@ -108,6 +129,31 @@ NOTE: Folders should only be created and used if they have content.
 │   │   │   ├── prod
 ----
 
+== Subcomponents
+Usually subcomponents have a single input for the viewmodel called `vm`. This viewmodel is a member of the components viewmodel and is build via selectors. If a subcomponent has also a subsubcomponent, the viewmodel of the subcomponent has a member containing the viewmodel of the subsubcomponent.
+
+== Dialogs
+There are two kinds of dialogs
+    * dialogs only used in one page
+    * dialogs used in multiple pages
+
+Both dialog types have in common that 
+* they are opened in effects
+* the result is handled in the effects and is converted into an action
+* they have a viewmodel
+* they are opened with `PortalDialogService.openDialog(...)`
+* they are communicating with the opener of the dialog by implementing `DialogResult`
+    * Implementing `DialogButtonClicked` can help setting the dialog result before the dialog closes
+
+=== Dialogs used in one page
+The components of this dialogs are subcomponents of the page they are used in. The subcomponents are placed in a subfolder called 'dialogs'. They do not have a septate section in the store because they are sharing the data of the page they are belonging to. The viewmodel can be passed via an input which is called 'vm'. The viewmodel is read from the store via a viewmodel selector in the effects or build with data from the action which is triggering the opening of the dialog.
```

Comment:
The viewmodel is read from the store via a viewmodel selector in the effects or _is_ build with data from the action which is triggering the opening of the dialog.